src/components/StepViews.tsx

```diff
@@ -43,6 +43,36 @@ export interface IStepViewsProps {
   views?: IViewProps[];
 }
 
+// @ts-ignore
+const api = (alertKaoto, index, saveConfig, replaceStep, step, stepInitialValues) => {
+  const kaotoApi: IStepExtensionApi = {
+    getCatalogSteps: getKaotoCatalogSteps,
+    getDeployment: getKaotoDeployment,
+    getDeploymentLogs: getKaotoDeploymentLogs,
+    getDeployments: getKaotoDeployments,
+    getDSLs: getKaotoDSLs,
+    getIntegrationJson: getKaotoIntegrationJson,
+    getIntegrationSource: getKaotoIntegrationSource,
+    getStep: () => {
+      return step;
+    },
+    getViews: getKaotoViews,
+    notifyKaoto: alertKaoto,
+    onKaotoButtonClicked,
+    saveConfig,
+    startDeployment: startKaotoDeployment,
+    step,
+    stepInitialValues,
+    stopDeployment: stopKaotoDeployment,
+    updateStep: (newStep: IStepProps) => {
+      // update state of step
+      replaceStep(newStep, index);
+    },
```

Comment:
I'd personally restrict the ap only to things required by stepExtensions.. which is probably just updateStep, and notifyKaoto. Also I'd rename the api to 'stepExtensionApi' and this save maybe for the general extension in future, maybe? 🤷‍♂️